### ISSUE TITLE
pip will still install pyside2 even without wheel on pypi

### ISSUE THREAD

**Issue by tlambert03** [napari/napari#1230]
## 🐛 Bug
as mentioned in #1229, it seems that pip pretty much always builds a wheel before installing (even if it downloads source), unless the user specifies `--no-cache-dir` ... which means that #1059 will not prevent pyside2 from being installed, even after #1215 .  If you `pip install napari>=0.3.0`, you will probably see it still build a wheel for caching.

You can repeat this locally:
```
# in napari repo
python setup.py sdist

# in clean environment
pip install path/to/napari/dist/napari-*.tar.gz 
...
Building wheels for collected packages: napari
  Building wheel for napari (PEP 517) ... done
  Created wheel for napari: filename=napari-0.3.0+6.gabe2afd6-py3-none-any.whl...
  Stored in directory: ~/Library/Caches/pip/wheels/...
Successfully built napari
...
# and only then does it build
Installing collected packages: ... napari

**Comment by sofroniewn** [comment]
I think @jni noticed this when testing the release candidate, but we weren't sure what was going on - this probably explains it

**Comment by tlambert03** [comment]
I asked a question about this on the [python packaging forum](https://discuss.python.org/t/conditional-package-install-depending-on-other-packages-in-environment/4140).  Got a very helpful answer there.  Long and short of it is that (as far as the responder knew) there's currently no good way to achieve what we want to do, and he suggested that living with `pip install napari[pyside2]`was probably our best option at this point.

He did provide useful links to two other very similar discussions that are going on.  [This one](https://github.com/pypa/setuptools/issues/1139) is particularly similar to what we want... it was mentioned by @ttung back in #704, but bears another look if you missed it the first time.

**Comment by Czaki** [comment]
Temporary fix is in #1246 (which disable isolated build), but until https://github.com/pypa/setuptools/issues/1139 is fixed there is no option for default extras.

**Comment by Czaki** [comment]
Maybe napari should define `all` extras which install all optional/default depedencies. Then if there is some other problem which implementation use then install command will not change.

**Comment by jni** [comment]
This was closed by #1324
